libraries/AP_Airspeed/AP_Airspeed_DLVR.cpp

```diff
@@ -104,18 +104,32 @@ void AP_Airspeed_DLVR::timer()
 
     if ((data >> STATUS_SHIFT)) {
         // anything other then 00 in the status bits is an error
-        Debug("DLVR: Bad status read %d", data >> STATUS_SHIFT);
+        Debug("DLVR: Bad status read %d", (int)(data >> STATUS_SHIFT));
         return;
     }
 
     uint32_t pres_raw = (data >> PRESSURE_SHIFT)    & PRESSURE_MASK;
     uint32_t temp_raw = (data >> TEMPERATURE_SHIFT) & TEMPERATURE_MASK;
 
     float press_h2o = 1.25f * 2.0f * range_inH2O * ((pres_raw - DLVR_OFFSET) / DLVR_SCALE);
+
+    if ((press_h2o > range_inH2O) || (press_h2o < -range_inH2O)) {
+        Debug("DLVR: Out of range pressure %f", press_h2o);
+        return;
+    }
+
     float temp = temp_raw * (200.0f / 2047.0f) - 50.0f;
 
     WITH_SEMAPHORE(sem);
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wfloat-equal" // suppress -Wfloat-equal as we are only worried about the case where we had never read
+    if ((temperature != 0) && (fabsf(temperature - temp) > 10)) {
```

Comment:
this is unsafe. If 1st packet is bad we would reject all future packets.
It is also unsafe for the sensor disconnecting then reconnecting some time later (with different temperature). 
To do this safely it has to start accepting again after a number of failures, or use a median filter

Reply:
I had considered a time out and originally rejected it, but you are right about both the first sample or really losing the sensor due to extreme noise. I've changed it to not apply this check if we have rejected more then 2 seconds of data. To reject 2 seconds of data we'd need to have ~100 readings be corrupted from the sensor. That does mean the temperature climb rate is now 5 degrees a second if you are experiencing this much corruption on the I2C bus, but given that this helps flag it out top you (and stops it from just using the bad data) I think that's fine.